app/browser/api/ledger.js

```diff
@@ -1800,25 +1806,32 @@ const generatePaymentData = (state) => {
       default:
         return
     }
-
-    try {
-      let chunks = []
-      qr.image(url, {type: 'png'})
-        .on('data', (chunk) => {
-          chunks.push(chunk)
-        })
-        .on('end', () => {
-          const paymentIMG = 'data:image/png;base64,' + Buffer.concat(chunks).toString('base64')
-          appActions.onLedgerQRGenerated(index, paymentIMG)
-        })
-    } catch (ex) {
-      console.error('qr.imageSync (for url ' + url + ') error: ' + ex.toString())
-    }
+    module.exports.qrWriteImage(index, url)
   })
 
   return state
 }
 
+const qrWriteImage = (index, url) => {
+  try {
+    let chunks = []
+    qr.image(url, {type: 'png'})
+      .on('data', (chunk) => {
+        chunks.push(chunk)
+      })
+      .on('end', () => {
+        const paymentIMG = 'data:image/png;base64,' + Buffer.concat(chunks).toString('base64')
```

Comment:
Let's move this one into `onLedgerQRGeneratedCallback` so that we have the whole logic there and we can just do  `.on('end', module.exports.onLedgerQRGeneratedCallback.bind(index, chunks))`

Reply:
++